Add a reworked dedicated page for "how to create a PR" (#471)
PrestaShop/docs: src/content/1.7/contribute/contribute-pull-requests/create-pull-request.md

---
title: Submit a Pull Request
weight: 4
---

# Publish your contribution on GitHub

> Once your changes sound good and tests pass on your local computer, you can submit these changes as a Pull Request on GitHub in order to contribute to the opensource project.

Reviewer: ```suggestion
Once your changes sound good and tests pass on your local computer, you can contribute to the open source project by submitting these changes as a Pull Request on GitHub.
```

First you need to push your branch on GitHub:

```
git push origin add-emoticons-support
```

> > You will need to use your GitHub credentials.

Reviewer: ```suggestion
{{% notice note %}}
You will need to use your GitHub credentials.
{{% /notice %}}
```

Now you can create your Pull Request on GitHub.

> If you do not know how to do it, you can read [GitHub documentation](https://help.github.com/articles/creating-a-pull-request/).

Reviewer: ```suggestion
Now you can create your Pull Request on GitHub. If you don't know how to do it, you can read [GitHub documentation](https://help.github.com/articles/creating-a-pull-request/).
```

> If you find this process quite complex, the following materials can help you (on dev.to):

Reviewer: ```suggestion
If you find this process quite complex, the following articles can help you:
```

I don't think it's useful to cite the source if you're linking to it.

> - [The SIMPLEST way to make a pull request](https://dev.to/lukegarrigan/the-simplest-way-to-make-a-pull-request-2h61)

Author: I added 2 helpful links

- [The github workflow explained](https://dev.to/mathieuks/introduction-to-github-fork-workflow-why-is-it-so-complex-3ac8)

> > Do not forget to complete the contribution table, this is really important for the Core Team to really understand what is the value of your contribution.

Reviewer: Maybe link to the contribution guidelines instead?

## Syncing your fork

PrestaShop Core is a really active project with more than 30 contributions accepted per week, so your copy will become outdated

> really fast. To make your own copy up to date with the original project, only a few commands are required:

Reviewer: ```suggestion
PrestaShop Core is a really active project with more than 30 contributions accepted per week, so your fork will become outdated really fast. To make your own copy up to date with the original project, only a few commands are required:
```

> > You need to execute these commands at the root of your copy/fork.

Reviewer: ```suggestion
{{% notice note %}}
You need to execute these commands at the root of your copy/fork.
{{% /notice %}}
```
